Makefile.base

```diff
@@ -67,10 +67,12 @@ $(BINDIR)/$(MODULE)/:
 
 $(BINDIR)/$(MODULE).a $(OBJ): | $(BINDIR)/$(MODULE)/
 
+relpath = $(shell realpath --relative-to=$(abspath .) $(1))
+
 $(BINDIR)/$(MODULE).a: $(OBJ) | $(DIRS:%=ALL--%)
 	@# Recreate archive to cleanup deleted/non selected source files objects
 	$(Q)$(RM) $@
-	$(Q)$(AR) $(ARFLAGS) $@ $^
+	$(Q)$(AR) $(ARFLAGS) $(foreach f,$(abspath $@ $^),$(call relpath,$f))
```

Comment:
instead of forking a new subshell we could patsubst away the BINDIR prefix from the objects, since the output (`$@`) is always located in the top of the BINDIR tree where all the objects are
```suggestion
	$(Q)$(AR) $(ARFLAGS) $@ $(patsubst $(dir $@)%,%,$^)
```

Reply:
The problem is that make is running from a different directory than BINDIR, so those paths will be wrong. Also "$@" also needs to be relative.

Reply:
I tried stripping RIOTBASE, but that will break external apps and external modules.

Reply:
Why does `$@` have to be relative?
I thought the relative paths were supposed to be relative to the archive file, I guess it needs a different approach.

Reply:
I played around a bit with the proposed makefile change.
Using this line I get fully working thin archives both in Docker and outside. `ar -t bin/frdm-kw41z/core.a` works on the host after building the archive inside Docker with `make BUILD_IN_DOCKER=1`, and the file names are shown as relative to the current directory instead of absolute.

```suggestion
	$(Q)cd $(dir $@) && $(AR) $(ARFLAGS) $@ $(patsubst $(dir $@)%,%,$^)
```

Reply:
We did some experiments with @cladmi on `ar` with relative and absolute paths. In all `ar` invocations, the paths given to the tool must make sense when interpreted from the current working directory.

- `ar rcTs ABSOLUTE.a ABSOLUTE.o` always works, but the path of the o file is stored as absolute in the a file.
- `ar rcTs  RELATIVE.a RELATIVE.o` always works, and `ar` stores the path of the o file relative to the a file (it converts it internally).
- `ar rcTs  ABSOLUTE.a RELATIVE.o` causes ar to store the path to the .o file **unchanged**, as it was given in the command line. This gives an invalid .a, unless the command is ran from the same directory where the a file is located.